src/hb-ot-name-table.hh

```diff
@@ -113,6 +114,32 @@ struct name
     return_trace (true);
   }
 
+  inline unsigned int get_count (void) const
+  { return count; }
+
+  inline const NameRecord& get_record(unsigned int index) const
+  { return hb_array_t<NameRecord> (nameRecordZ.arrayZ, count)[index]; }
+
+  inline unsigned int get_record_string(unsigned int  index,
```

Comment:
I think from this function we want to return a hb_bytes_t.  The API entrypoint can decide whether to return it as is, or to copy, or to transcribe it to UTF-8 or UTF-16.